File: 50 Coding Challenge Part I/Python/Challenge24.py
"""
Challenge 24: Create a function that will merge
two arrays and return the result as
a new array
"""


def mergeArray(array1, array2):
    """empty array to hold the new values"""
    new_array = []
    for i in array1:
        new_array.append(i)
    for j in array2:
        new_array.append(j)
    return sorted(new_array)


# Driver Method
arr1 = [1, 4, 5, 7, 3, 8]
arr2 = [34, 23, 344, 86]
print(mergeArray(arr1, arr2))

File: 50 Coding Challenge Part I/JavaScript/Challenge24.js
// Challenge 24: Create a function that will merge
// two arrays and return the result as a new array

function mergeArray(arr1, arr2){
    let newArr = [];

    for(let arr of arr1){
        newArr.push(arr);
    }
    for(let arr of arr2){
        newArr.push(arr);
    }

    return newArr;
}

// Using the .concat() method
const addArray = (array1, array2) => array1.concat(array2);

// Driver Method
let ar1 = [1, 2, 3, 4];
let ar2 = [5, 6, 7, 8];
console.log(mergeArray(ar1, ar2));
// used another method
console.log(addArray(ar1, ar2));
